### ISSUE TITLE
LDAP and Local Logins

### ISSUE THREAD

**Issue by vish84** [Ylianst/MeshCentral#1632]
Hi

I am looking at setting up LDAP for users to login. I have this working to a degree, but noticed that once this was configured I was no longer able to login using the local admin account. I originally set my server up without any LDAP and the first account was admin. I found a work around, of disabling the LDAP config, and then giving permissions to a domain user who just logged in to have admin. However would be nice to be able to login with either local accounts or domain accounts. As i'm trying to do some automation and prefer to have an admin account not associated with a user. I know I can use an LDAP service account, but preference would be to keep this local. 

i'm unsure if this can be done at present or not. 

Many thanks

**Comment by Tomatoschewps** [comment]
@supr4s ?

**Comment by supr4s** [comment]
Hi everyone,

In my case, after set up LDAP connection in config.json and restart meshcentral systemd service, the account creation option was again available in the web interface. So I was able to create a local administrator account in addition to the LDAP connection.

**Comment by vish84** [comment]
Thank @supr4s, the option to add a new account is available to me in the UI but when I try to create an account I encounter an error New Account: Unable to add user in this mode.

**Comment by supr4s** [comment]
In our case, after various tests, i'd lost access to web interface with the local account.
LDAP authentication was active 15 minutes ago, and i could connect with a local account at the same time without problem.
That's a really weird behavior !

I disabled LDAP authentication in settings.json, restart meshcentral service => still no connection with local account.
Forced to restart the server => good, authentication with local account work fine ! 

I have re-enabled LDAP authentication, restarting the service => no more possible to log with a local account.

In summary, I think it is not possible to have LDAP authentication and local account authentication at the same time. However, it should be the case, so as not to be blocked if the LDAP server is not functional.

**Comment by vish84** [comment]
Correct this is the same behaviour I was seeing, as mentioned in my first post.
Would be good to be able to login with a local account at the same time

**Comment by informaticavera** [comment]
I agree, the system should provide the two authentication methods, local (bd) and ldap.

There are new options like login token that do not work with active ldap authentication.

**Comment by si458** [comment]
sadly this isnt possible because the unique user identifiers will be the same for ldap and local
`user//simon` -> `user/domain/user`
so if u setup accounts locally, then switch to ldap, ldap will just use the local accounts permission/groups, etc that have already been created
